server/bleep/src/agent/tools/code.rs

```diff
@@ -1,29 +1,42 @@
 use anyhow::Result;
-use tracing::{debug, info, instrument, trace};
+use tracing::{debug, info, trace};
 
 use crate::{
     agent::{
-        exchange::{CodeChunk, SearchStep, Update},
-        prompts, Agent,
+        exchange::{CodeChunk, RepoPath, SearchStep, Update},
+        prompts, Agent, SemanticSearchParams,
     },
     analytics::EventData,
     llm_gateway,
+    query::parser::Literal,
 };
 
 impl Agent {
-    #[instrument(skip(self))]
-    pub async fn code_search(&mut self, query: &String) -> Result<String> {
+    pub async fn code_search(&mut self, query: &str) -> Result<String> {
         const CODE_SEARCH_LIMIT: u64 = 10;
         const MINIMUM_RESULTS: usize = CODE_SEARCH_LIMIT as usize / 2;
 
         self.update(Update::StartStep(SearchStep::Code {
-            query: query.clone(),
+            query: query.to_owned(),
             response: String::new(),
         }))
         .await?;
 
+        // not sure if this is what we want here, or use the project coupled with the agent.
+        //
+        // let repos = self.paths().map(|r| r.repo.to_string()).collect::<Vec<_>>();
+        let repos = self.project.clone();
+
         let mut results = self
-            .semantic_search(query.into(), vec![], CODE_SEARCH_LIMIT, 0, 0.3, true)
+            .semantic_search(SemanticSearchParams {
+                query: Literal::from(&query.to_string()),
+                paths: vec![],
+                project: repos.clone(),
+                limit: CODE_SEARCH_LIMIT,
+                offset: 0,
```

Comment:
Could make `offset` and `retrieve_more` default params here.

Reply:
I wanted it easier to understand what the different parameters are. Unfortunately we can't `Default` this struct, so we need to pass these every time. To fix this we'd need to restructure the way this function works, or which component it belongs to.